compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -161,8 +161,8 @@ object DesugarEnums {
       val (tagMeth, scaffolding) = enumTagMeth(CaseKind.Object)
       val impl1 = cpy.Template(impl)(body =
         impl.body ++ List(tagMeth, toStringMeth) ++ registerCall)
-      val vdef = ValDef(name, TypeTree(), New(impl1)).withMods(mods | Final).withPos(pos)
-      flatTree(scaffolding ::: vdef :: Nil).withPos(pos.startPos)
+      val vdef = ValDef(name, TypeTree(), New(impl1)).withMods(mods | Final)
+      flatTree(scaffolding ::: vdef :: Nil).withPos(pos)
```

Comment:
Can you add an explanation how that makes a difference? I thought that the previous version would give `pos`
to the actual value definition and `startPos` to scaffolding. Isn't that what we want?